### PR TITLE
clickhouse-client: don't show "0 rows in set" if it is zero and if exception was thrown

### LINKED ISSUE #52936: clickhouse-client: don't show "0 rows in set" if it is zero and if exception was thrown.

### PR DESCRIPTION
**Pull request by aiven-sal**
Close #52936

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
clickhouse-client won't show "0 rows in set" if it is zero and if exception was thrown

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 5d2a80c9f1f4f96e2d1794ba75d0d2bde1205116 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55240/5d2a80c9f1f4f96e2d1794ba75d0d2bde1205116/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6407377901">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55240/5d2a80c9f1f4f96e2d1794ba75d0d2bde1205116/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Client/ClientBase.cpp [hunk 1]
```diff
@@ -1879,9 +1879,10 @@ void ClientBase::processParsedSingleQuery(const String & full_query, const Strin
 
     if (is_interactive)
     {
-        std::cout << std::endl
-            << processed_rows << " row" << (processed_rows == 1 ? "" : "s")
-            << " in set. Elapsed: " << progress_indication.elapsedSeconds() << " sec. ";
+        std::cout << std::endl;
+        if (!server_exception)
```
Comment (alexey-milovidov) [inline src/Client/ClientBase.cpp:1883]: The logic is slightly different than was proposed

Code:

> don't show "... rows in set" if exception was thrown

Proposed:

> don't show "0 rows in set" if it is zero and if exception was thrown

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

**Comment by alexey-milovidov** [comment]
As it is a user-visible change, let's add it to the changelog.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Could you please merge or rebase with master?

**Pull request merged**